provider/testdata/benches/providers.rs

```diff
@@ -1,117 +0,0 @@
-// This file is part of ICU4X. For terms of use, please see the file
-// called LICENSE at the top level of the ICU4X source tree
-// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
-
-extern crate alloc;
-
-use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use icu_locid::locale;
-use icu_provider::hello_world::HelloWorldV1Marker;
-use icu_provider::prelude::*;
-use icu_provider::AsDeserializingBufferProvider;
-use icu_provider_blob::BlobDataProvider;
-
-static POSTCARD_BYTES: &[u8] = include_bytes!(concat!(
-    env!("CARGO_MANIFEST_DIR"),
-    "/data/testdata.postcard"
-));
-
-struct BakedDataProvider;
-struct BakedAnyProvider;
-
-mod baked {
-    include!(concat!(env!("CARGO_MANIFEST_DIR"), "/data/baked/mod.rs"));
-    impl_data_provider!(super::BakedDataProvider);
-    impl_any_provider!(super::BakedAnyProvider);
-}
-
-#[inline(never)]
-fn create_static_data_provider() -> BlobDataProvider {
```

Comment:
Issue: These benches were added to benchmark the "real-world" constructor time (acknowledging that testdata doesn't have all locales, but it certainly is much closer to real-life than the helloworld-only provider).

Reply:
Ack. However, in addition to testdata not having all locales, it does have all keys, which isn't real world if you care about performance.

I think we should split this into multiple benches:
- many keys (fixed set) no locales: benches the key lookup
- one key many locales: benches the locale lookup
- one key one locale: benches the deserialization

Currently we test many keys with 4 locales, but the key set is not stable, and 4 locales don't really exercise the locale lookup. It'd be nice for this benchmark to only be affected by changes to the infrastructure, not to testdata.

Using a fixed blob instead of one changing with testdata can also serve as a canary for serde stability. We'd generate the blobs with datagen 1.0 and minimal deps.

Reply:
Acknowledged. However, we should not delete this bench, our only bench on non-trivial data, until a replacement is ready.

Also, regarding many-keys-few-locales, this is something that could happen in real life in either of the following scenarios:

1. Baked data comes with a few hand-chosen locales
2. A dynamically loaded language pack contains all keys for one locale or one language with regional variants